protocol/lib/tx_mode.go

```diff
@@ -12,7 +12,10 @@ func AssertDeliverTxMode(ctx sdk.Context) {
 }
 
 func IsDeliverTxMode(ctx sdk.Context) bool {
-	return !ctx.IsCheckTx() && !ctx.IsReCheckTx()
+	if ctx.IsCheckTx() || ctx.IsReCheckTx() || ctx.ExecMode() == sdk.ExecModeCheck || ctx.ExecMode() == sdk.ExecModeReCheck {
```

Comment:
`InitGenesis` runs in `ExecMode` == `ModeCheckTx` but `IsCheckTx == false` for some reason.

We don't want to stage events during `InitGenesis`, because:
- InitGenesis stages the events.
- HydrateClobPair (Preblocker) sees the ClobPair objects and create in memory object
- In Precommit, we attempt to create the `Clob` objects again, but fail because it's already done above.